### PR TITLE
Recovered RL tests. RateLimit package refactor

### PR DESCRIPTION
**Pull request by RodrigoAD**

### CODE REVIEW

File: packages/core/bootstrap/src/lib/rate-limit/config.ts [hunk 1]
```diff
@@ -13,15 +13,24 @@ export interface Config {
...
+    const provider = getEnv('API_PROVIDER') || ''
+    const tier = getEnv('API_TIER') || ''
+    const providerConfig = getRateLimit(provider, tier)
+    capacity = Number(providerConfig?.quota)
+  }
   return {
     hashOpts: getHashOpts(),
-    totalCapacity: capacity || providerConfig?.quota,
+    totalCapacity: capacity,
+    enabled: !!capacity,
```
Comment (krebernisak) [inline packages/core/bootstrap/src/lib/rate-limit/config.ts:34]: Could you please make an `EXPERIMENTAL_RATE_LIMIT_ENABLED` var so we are more explicit about it.

While doing so, please also add the `EXPERIMENTAL_` prefix to the `WARMUP_ENABLED` var :)

File: packages/core/ratelimits/src/index.ts [hunk 2]
```diff
@@ -38,32 +38,43 @@ const calculateLimits = (declaredTier: DeclaredTier) => {
...
-    : (Limits[provider] as DeclaredTier[])?.find((element) => element.tierName === tier)
-  if (matchedTier === undefined) {
-    console.log(
-      `[provider, tier]: [${provider}, ${tier}] doesn't match any provider spec in limits.json`,
-    )
+const findTier = (providerName: string, tier: number | string): DeclaredTier | undefined => {
+  const tierIndex = Number(tier)
+  const provider: DeclaredTier[] = Limits[providerName.toLowerCase()]
+  if (!provider) {
+    console.log(`Rate Limit: Provider with name ${providerName} not found`)
```
Comment (krebernisak) [inline packages/core/ratelimits/src/index.ts:48]: Use out logger instead of `console.log`

File: packages/core/ratelimits/src/index.ts [hunk 3]
```diff
@@ -38,32 +38,43 @@ const calculateLimits = (declaredTier: DeclaredTier) => {
...
-    declaredTier !== undefined
-      ? calculateLimits(declaredTier)
-      : [config.DEFAULT_SECOND_RATELIMIT, config.DEFAULT_MINUTE_RATELIMIT]
-  return {
-    burst: burst,
-    quota: quota,
-    second: burst,
-    minute: quota,
+  if (!declaredTier) {
+    console.log(
```
Comment (krebernisak) [inline packages/core/ratelimits/src/index.ts:63]: Use out logger instead of `console.log`

**Review by krebernisak — CHANGES_REQUESTED** [review]

File: packages/core/ratelimits/src/index.ts (hunk 2 above)
Comment (RodrigoAD) [reply packages/core/ratelimits/src/index.ts]: It will raise circular dependency problems...

Comment (justinkaseman) [reply packages/core/ratelimits/src/index.ts:48]: It shouldn't. Logger is internal to bootstrap now.

File: packages/core/bootstrap/src/lib/rate-limit/config.ts [hunk 4]
```diff
@@ -13,15 +13,25 @@ export interface Config {
...
+   */
+  enabled: boolean
 }
 
 export function get(): Config {
-  const capacity = parseInt(getEnv('RATE_LIMIT_CAPACITY') || '')
-  const provider = getEnv('API_PROVIDER') || ''
-  const tier = parseInt(getEnv('API_TIER') || '')
-  const providerConfig = getRateLimit(provider, tier)
+  const enabled = parseBool(getEnv('EXPERIMENTAL_RATE_LIMIT_ENABLED'))
```
Comment (justinkaseman) [inline packages/core/bootstrap/src/lib/rate-limit/config.ts:24]: Thanks for getting this change in :)

File: packages/core/ratelimits/src/index.ts (hunk 2 above)
Comment (justinkaseman) [reply packages/core/ratelimits/src/index.ts:48]: He's right about this. Maybe we should move `ratelimit` stuff into `bootstrap`.

File: packages/core/bootstrap/README.md [hunk 5]
```diff
@@ -11,8 +11,14 @@ A metrics server can be exposed which returns prometheus compatible data on the
 - `METRICS_PORT`: Optional number, defaults to `9080`, set to change the port the `/metrics` endpoint is served on
 - `METRICS_NAME`: Optional string, defaults to 'N/A', set to apply a label of `NAME` to each metric
 
+## Rate Limit
+To avoid hitting rate limit issues with the data provider subscription, a rate limit capacity per minute can be set:  
+- Option 1, manual capacity setting:
+   - `RATE_LIMIT_CAPACITY`: Maximum capacity on requests per minute
+- Option 2, capacity by reference. Check your plan [here]('../ratelimits/src/limits.json') and use it with the following configuration:
+   - `API_PROVIDER`: Name of the provider
+   - `API_TIER`: Plan you are subscribed to
```
[the remark this answers is not in the document]
Comment (justinkaseman) [reply packages/core/bootstrap/README.md]: The framework doesn't currently know what adapter it is running on. How do you feel about passing `NAME` through to `bootstrap.expose()`?

File: packages/core/ratelimits/src/index.ts (hunk 2 above)
Comment (krebernisak) [reply packages/core/ratelimits/src/index.ts:48]: There is probably no reason for any logging to happen in `ratelimits`, it's just a helper package. Whoever uses it should do the logging.

**Review by krebernisak — APPROVED** [review]

**Pull request merged**